From: oulijun <oulijun@huawei.com>
Date: Fri, 4 May 2018 10:57:10 +0800
Subject: RDMA/hns: Set desc_dma_addr for zero when free cmq desc
Patch-mainline: v4.17-rc7
Git-commit: 90e7a4d5066240b75cdfd1bf8944ca36622153b1
References: bsc#1104427 FATE#326416

In order to avoid illegal use for desc_dma_addr of ring,
it needs to set it zero when free cmq desc.

Signed-off-by: Lijun Ou <oulijun@huawei.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/hns/hns_roce_hw_v2.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -617,6 +617,8 @@ static void hns_roce_free_cmq_desc(struc
 	dma_unmap_single(hr_dev->dev, ring->desc_dma_addr,
 			 ring->desc_num * sizeof(struct hns_roce_cmq_desc),
 			 DMA_BIDIRECTIONAL);
+
+	ring->desc_dma_addr = 0;
 	kfree(ring->desc);
 }
 
